Frameworks/Foundation/NSDictionary.mm

```diff
@@ -99,14 +99,14 @@ - (NSArray*)allObjects {
         return nil;
     }
 
-    StrongId<NSArray> keyArray = [_keyEnumerator allObjects];
-    StrongId<NSMutableArray> toRet = [NSMutableArray new];
+    NSArray* keyArray = [_keyEnumerator allObjects];
+    NSMutableArray* toRet = [NSMutableArray array];
 
-    for (id key in static_cast<NSArray*>(keyArray)) {
+    for (id key in keyArray) {
```

Comment:
Actually, there's a pointless copy here!
NSEnumerator can be fast-enumerated using `for(...in...)` syntax just as well as an array can.

We can kill the temporary, the retain on all the keys, and all their various friends by using `for (id key in _keyEnumerator)`.